homeassistant/components/saj/sensor.py

```diff
@@ -40,24 +45,43 @@
     "": None,
 }
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_HOST): cv.string})
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_TYPE, default=INVERTER_TYPES[0]): vol.In(INVERTER_TYPES),
+        vol.Optional(CONF_USERNAME, default=""): cv.string,
+        vol.Optional(CONF_PASSWORD, default=""): cv.string,
```

Comment:
You can use `vol.Inclusive` to make sure username and password are always specified together.

The default doesn't make much sense here, so drop it.

Reply:
Changed it, thanks!